diff --git a/bin/find_member b/bin/find_member
index e772738..99e4ee6 100755
--- a/bin/find_member
+++ b/bin/find_member
@@ -76,7 +76,7 @@ def usage(code, msg=''):
         fd = sys.stderr
     else:
         fd = sys.stdout
-    print >> fd, _(__doc__)
+    print >> fd, C_(__doc__)
     if msg:
         print >> fd, msg
     sys.exit(code)
diff --git a/bin/genaliases b/bin/genaliases
index 931f2bd..1a102cf 100644
--- a/bin/genaliases
+++ b/bin/genaliases
@@ -55,7 +55,7 @@ def usage(code, msg=''):
         fd = sys.stderr
     else:
         fd = sys.stdout
-    print >> fd, _(__doc__)
+    print >> fd, C_(__doc__)
     if msg:
         print >> fd, msg
     sys.exit(code)
diff --git a/bin/mmsitepass b/bin/mmsitepass
index c479e2e..bb4cc71 100755
--- a/bin/mmsitepass
+++ b/bin/mmsitepass
@@ -54,7 +54,7 @@ def usage(code, msg=''):
         fd = sys.stderr
     else:
         fd = sys.stdout
-    print >> fd, _(__doc__)
+    print >> fd, C_(__doc__)
     if msg:
         print >> fd, msg
     sys.exit(code)
diff --git a/bin/reset_pw.py b/bin/reset_pw.py
index cc35cd7..b71d532 100644
--- a/bin/reset_pw.py
+++ b/bin/reset_pw.py
@@ -38,7 +38,7 @@ import getopt
 
 import paths
 from Mailman import Utils
-from Mailman.i18n import _C
+from Mailman.i18n import C_
 
 
 try:
@@ -54,7 +54,7 @@ def usage(code, msg=''):
         fd = sys.stderr
     else:
         fd = sys.stdout
-    print >> fd, _(__doc__.replace('%', '%%'))
+    print >> fd, C_(__doc__.replace('%', '%%'))
     if msg:
         print >> fd, msg
     sys.exit(code)
