diff --git a/Mailman/ListAdmin.py b/Mailman/ListAdmin.py
index ec1307a..321d9f6 100644
--- a/Mailman/ListAdmin.py
+++ b/Mailman/ListAdmin.py
@@ -202,7 +202,7 @@ class ListAdmin:
                     cPickle.dump(msg, fp, 1)
                 else:
                     g = Generator(fp)
-                    g(msg, 1)
+                    g.flatten(msg, 1)
                 fp.flush()
                 os.fsync(fp.fileno())
             finally:
@@ -251,7 +251,7 @@ class ListAdmin:
             outfp = open(outpath, 'w')
             try:
                 g = Generator(outfp)
-                g(msg, 1)
+                g.flatten(msg, 1)
             finally:
                 outfp.close()
         # Now handle updates to the database
