diff --git a/bin/newlist b/bin/newlist
index 3a3fd95..1d72cd5 100755
--- a/bin/newlist
+++ b/bin/newlist
@@ -104,6 +104,7 @@ from Mailman import Message
 from Mailman import i18n
 
 C_ = i18n.C_
+_ = i18n._
 
 PROGRAM = sys.argv[0]
 
@@ -247,7 +248,7 @@ def main():
         try:
             msg = Message.UserNotification(
                 owner_mail, siteowner,
-                C_('Your new mailing list: %(listname)s'),
+                _('Your new mailing list: %(listname)s'),
                 text, mlist.preferred_language)
             msg.send(mlist)
         finally:
